Build of 3.5.3 fails when both ACL and graph are enabled

# Summary
Build of 3.5.3 fails when both ACL and graph are enabled, due to usage of internal ACL headers.
List of exported headers: https://github.com/ARM-software/ComputeLibrary/blob/main/SConstruct#L646
Error log:
```
[  240s] [  7%] Building CXX object src/graph/backend/dnnl/CMakeFiles/dnnl_graph_backend_dnnl.dir/platform.cpp.o
[  240s] cd /home/abuild/rpmbuild/BUILD/oneDNN-3.5.3/build/src/graph/backend/dnnl && /usr/bin/c++ -DDNNL_AARCH64=1 -DDNNL_AARCH64_USE_ACL -DDNNL_DLL -DDNNL_DLL_EXPORTS -DDNNL_ENABLE_CPU_ISA_HINTS -DDNNL_ENABLE_ITT_TASKS -DDNNL_ENABLE_MAX_CPU_ISA -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/home/abuild/rpmbuild/BUILD/oneDNN-3.5.3/build/include -I/home/abuild/rpmbuild/BUILD/oneDNN-3.5.3/include -I/home/abuild/rpmbuild/BUILD/oneDNN-3.5.3/src -mbranch-protection=standard -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -fopenmp -fvisibility-inlines-hidden  -Wall -Wno-unknown-pragmas -fvisibility=internal -O3 -mcpu=generic  -fPIC -Wformat -Wformat-security -fstack-protector-strong  -Wmissing-field-initializers  -Wno-strict-overflow -Wno-maybe-uninitialized  -O2 -g -DNDEBUG -std=c++14 -MD -MT src/graph/backend/dnnl/CMakeFiles/dnnl_graph_backend_dnnl.dir/platform.cpp.o -MF CMakeFiles/dnnl_graph_backend_dnnl.dir/platform.cpp.o.d -o CMakeFiles/dnnl_graph_backend_dnnl.dir/platform.cpp.o -c /home/abuild/rpmbuild/BUILD/oneDNN-3.5.3/src/graph/backend/dnnl/platform.cpp
[  240s] /home/abuild/rpmbuild/BUILD/oneDNN-3.5.3/src/graph/backend/dnnl/platform.cpp:27:10: fatal error: src/common/cpuinfo/CpuInfo.h: No such file or directory
[  240s]    27 | #include "src/common/cpuinfo/CpuInfo.h"
[  240s]       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[  240s] compilation terminated.
[  240s] make[2]: *** [src/graph/backend/dnnl/CMakeFiles/dnnl_graph_backend_dnnl.dir/build.make:177: src/graph/backend/dnnl/CMakeFiles/dnnl_graph_backend_dnnl.dir/platform.cpp.o] Error 1
[  240s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/oneDNN-3.5.3/build'
[  240s] make[2]: *** Waiting for unfinished jobs....
```

# Version
oneDNN 3.5.3 + patch #2128 

# Environment
* CPU make and model: aarch64
* OS version (`uname -a`) Linux (openSUSE Tumbleweed)
* Compiler version (`gcc --version`) 14.2.0
* CMake version (`cmake --version`) 3.30.3
* CMake output log
* git hash (`git log -1 --format=%H`)

# Steps to reproduce
Enable ACL + graph on openSUSE Tumbleweed aarch64


# Observed behavior
Build fails.

# Expected behavior
Build should succeed by using only exported headers from ACL.

--- a/.github/automation/build_aarch64.sh
+++ b/.github/automation/build_aarch64.sh
@@ -30,17 +30,24 @@ export ACL_ROOT_DIR=${ACL_ROOT_DIR:-"${PWD}/ComputeLibrary"}
 
 CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-"Release"}
 ONEDNN_TEST_SET=SMOKE
+ONEDNN_BUILD_GRAPH=1
 
 # ACL is not built with OMP on macOS.
 if [[ "$OS" == "Darwin" ]]; then
     ONEDNN_THREADING=SEQ
+    if [[ "$BUILD_TOOLSET" == "clang" ]]; then
+        if [[ "$CMAKE_BUILD_TYPE" == "Debug" ]]; then
+            # Darwin graph tests take a lot of time in debug mode.
+            ONEDNN_BUILD_GRAPH=0
+        fi
+    fi
 fi
 
 set -x
 cmake \
     -Bbuild -S. \
     -DDNNL_AARCH64_USE_ACL=ON \
-    -DONEDNN_BUILD_GRAPH=0 \
+    -DONEDNN_BUILD_GRAPH=$ONEDNN_BUILD_GRAPH \
     -DDNNL_CPU_RUNTIME=$ONEDNN_THREADING \
     -DONEDNN_WERROR=ON \
     -DDNNL_BUILD_FOR_CI=ON \

--- a/.github/automation/test_aarch64.sh
+++ b/.github/automation/test_aarch64.sh
@@ -62,9 +62,21 @@ if [[ "$OS" == "Linux" ]]; then
         SKIPPED_TEST_FAILURES+="|test_benchdnn_modeC_conv_smoke_cpu"
         SKIPPED_TEST_FAILURES+="|test_benchdnn_modeC_deconv_smoke_cpu"
         SKIPPED_TEST_FAILURES+="|test_benchdnn_modeC_matmul_smoke_cpu"
+        SKIPPED_TEST_FAILURES+="|cpu-graph-gqa-cpp"
+        SKIPPED_TEST_FAILURES+="|cpu-graph-mqa-cpp"
+        SKIPPED_TEST_FAILURES+="|cpu-graph-sdpa-cpp"
+        SKIPPED_TEST_FAILURES+="|cpu-graph-sdpa-stacked-qkv-cpp"
+        SKIPPED_TEST_FAILURES+="|test_graph_unit_dnnl_large_partition_usm_cpu"
+        SKIPPED_TEST_FAILURES+="|test_graph_unit_dnnl_sdp_decomp_usm_cpu"
     elif [[ "$CMAKE_BUILD_TYPE" == "Release" ]]; then
         SKIPPED_TEST_FAILURES="cpu-primitives-deconvolution-cpp"
         SKIPPED_TEST_FAILURES+="|test_benchdnn_modeC_lnorm_smoke_cpu"
+        SKIPPED_TEST_FAILURES+="|cpu-graph-gqa-cpp"
+        SKIPPED_TEST_FAILURES+="|cpu-graph-mqa-cpp"
+        SKIPPED_TEST_FAILURES+="|cpu-graph-sdpa-cpp"
+        SKIPPED_TEST_FAILURES+="|cpu-graph-sdpa-stacked-qkv-cpp"
+        SKIPPED_TEST_FAILURES+="|test_graph_unit_dnnl_large_partition_usm_cpu"
+        SKIPPED_TEST_FAILURES+="|test_graph_unit_dnnl_sdp_decomp_usm_cpu"
     fi
 elif [[ "$OS" == "Darwin" ]]; then
     if [[ "$CMAKE_BUILD_TYPE" == "Debug" ]]; then
@@ -85,7 +97,7 @@ if [[ "$OS" == "Darwin" ]]; then
 elif [[ "$OS" == "Linux" ]]; then
     if [[ "$ONEDNN_THREADING" == "OMP" ]]; then
         # OMP is already multi-threaded. Let's not oversubscribe.
-        CTEST_MP=-j2
+        CTEST_MP=-j3
     elif [[ "$ONEDNN_THREADING" == "SEQ" ]]; then
         CTEST_MP=$MP
     fi

--- a/src/graph/backend/dnnl/kernels/sdp_primitive_config.cpp
+++ b/src/graph/backend/dnnl/kernels/sdp_primitive_config.cpp
@@ -15,6 +15,7 @@
 *******************************************************************************/
 
 #include "graph/backend/dnnl/kernels/sdp_primitive_config.hpp"
+#include "common/compiler_workarounds.hpp"
 
 namespace dnnl {
 namespace impl {

